### PR TITLE
refactor: removed 1.12.0 deprecated fields.

### PR DESCRIPTION
**Pull request by marcosuma**
## Description

Removing all the fields that have been declared as to be deprecated in v1.12.0.

Link to any related issue(s): https://jira.mongodb.org/browse/INTMDB-1015

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [X] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [X] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [X] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [X] I have run make fmt and formatted my code

### CODE REVIEW

**Review by andreaangiolillo — COMMENTED** [review]
Make sure to update the documentation as well: https://github.com/mongodb/terraform-provider-mongodbatlas/tree/master/website/docs

r = resources and d = data sources

File: mongodbatlas/resource_mongodbatlas_cloud_provider_access_setup_test.go
Comment (AgustinBettati) [inline mongodbatlas/resource_mongodbatlas_cloud_provider_access_setup_test.go]: question: why was this test deleted?

Comment (andreaangiolillo) [reply mongodbatlas/resource_mongodbatlas_cloud_provider_access_setup_test.go]: good catch. This acceptance test is testing the import function so it should still be relevant

**Review by andreaangiolillo — COMMENTED** [review]
changes LGTM after reverting the removed acceptance tests. 

Great work here 💯 !

One last action to do as part of this ticket is to add a list of deprecated fields per resource/datasource in the PR description (or in the JIRA ticket) to help us add this info in the CHANGELOG before the release.

File: mongodbatlas/resource_mongodbatlas_cloud_provider_access_setup_test.go
Comment (marcosuma) [reply mongodbatlas/resource_mongodbatlas_cloud_provider_access_setup_test.go]: my bad. I was biased by these lines
```
					resource.TestCheckResourceAttrSet(resourceName, "aws.atlas_assumed_role_external_id"),
					resource.TestCheckResourceAttrSet(resourceName, "aws.atlas_aws_account_arn"),
```
which are not checking the import function and they are checking `aws` field that does not exist anymore. Let me re-add the test and fix the expectations. Thank you @AgustinBettati

**Review by AgustinBettati — APPROVED** [review]
LGTM, thanks for addressing the test change. Would look into Andreas last comment before merging.

**Pull request merged**
